### PR TITLE
Replace syntax highlighter with Highlight.js

### PR DESCRIPTION
**Pull request by rshest**
This PR swaps the old shjs syntax highlighter with [Highlight.js](https://highlightjs.org/) one.

The latter supports modern C++ and is more widely used/well maintained.

I have embedded Highlight.js v9.12.0 with the "VS" color theme, since it looked closest to what it was before, visually (having removed all the other, unused themes). 

Please take a look at the [Highlight.js demo here](https://highlightjs.org/static/demo/) and see if there is a color theme that may look slicker/more modern - we could change the default one, then (or, ultimately, have color theme as a configurable option, but this is beyond the scope of this PR).

Resolves https://github.com/knatten/cppquiz/issues/104

### CODE REVIEW

File: templates/quiz/categorize.html [hunk 1]
```diff
@@ -6,8 +6,8 @@ <h2>Categorize difficulties</h2>
 <tr class="difficulty_{{q.difficulty}}">
     <td>
         <a id="question_{{q.pk}}"/>
-        <pre class="sh_cpp">
-{{q.question}}
+        <pre>
+            <code class="cpp hljs">{{q.question}}</code>
```
Comment (knatten) [inline templates/quiz/categorize.html]: The added `code` inside the `pre` results in a bit more margin than we used to have, but I guess the site could use some whitespace anyway! :)

Comment (rshest) [reply templates/quiz/categorize.html]: The reason why we have both `<pre>` and `<code>` there is that Highlight.js by default treats this combination as code blocks to process (but [that could be customized](https://github.com/isagalaev/highlight.js#custom-initialization)).

I could add something like a "code-container" class on the outer `<pre>` there, and fiddle with the margins in CSS on that class, what do you think?

**Comment by rshest** [comment]
Thanks! 

> Now brace for the bike shedding..

Well, you see, my own first choice was actually exactly the Github one, but the background color did not seem to fit well together with the rest of the stuff... so I guess this is the rare case when bike shedding comes to unanonymous agreement regarding the bike shed color, from the start :)

**Comment by rshest** [comment]
My "Github" seems to be looking a little bit different, though - I wonder if this is just because of the different default settings (mine are Chrome and Firefox on Linux Mint 18.1), or is it a different Github color scheme version?

image

As a side note, I wonder if something like [reset.css](https://meyerweb.com/eric/tools/css/reset/) would be beneficial to normalize things like these?

**Comment by knatten** [comment]
Haha, that's funny! :D 

I think the differences are just due to the font renderer. The fact that the letters get a bit fatter on my Chrome/macOS seems to also make them look darker. (I tried picking out some of the actual letter colors in the screen shot, and they seem to be roughly the same.)

I don't think we need to bother with reset.css.

File: templates/quiz/categorize.html (hunk 1 above)
Comment (knatten) [reply templates/quiz/categorize.html]: Nah, don't bother. I'm planning to adjust the margins a bit in other places too.

**Pull request merged**